doc/statistics/z_test.qbk

```diff
@@ -0,0 +1,114 @@
+[/
+Copyright (c) 2021 Matt Borland
+Use, modification and distribution are subject to the
+Boost Software License, Version 1.0. (See accompanying file
+LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+]
+
+[section:z_test /z/-tests]
+
+[heading Synopsis]
+
+```
+#include <boost/math/statistics/z_test.hpp>
+
+namespace boost::math::statistics {
+
+template<typename Real, typename Size = Real>
```

Comment:
Is `typename Size = Real` a good naming convention here? I feel like this default may be incorrect.

Reply:
It is incorrect. I was toying with not having to call `Real(sample_size)` but removed it.